lib/logstash/filters/json.rb

```diff
@@ -46,54 +46,64 @@ class LogStash::Filters::Json < LogStash::Filters::Base
   # NOTE: if the `target` field already exists, it will be overwritten!
   config :target, :validate => :string
 
-  public
+  JSONPARSEFAILURE_TAG = "_jsonparsefailure"
+
   def register
     # Nothing to do here
-  end # def register
+  end
 
-  public
   def filter(event)
-    
-
-    @logger.debug("Running json filter", :event => event)
-
-    return unless event.include?(@source)
-
-    # TODO(colin) this field merging stuff below should be handled in Event.
+    @logger.debug? && @logger.debug("Running json filter", :event => event)
 
     source = event[@source]
+    return unless source
```

Comment:
In the old code if `event[@source]` was nil then `LogStash::Json.load(nil)` blows up (we should fix this).  So the new impl is better.  However, what is the expectation on the value of `event[@target]` for downstream filters? Nil or empty string?